diff --git a/CRC-HL2_EmitaCityRP.Emita_2025/spawnLootSystem.sqf b/CRC-HL2_EmitaCityRP.Emita_2025/spawnLootSystem.sqf
index c16ddb88818f545332152a8f8bce5ab3f07c3e62..8d17697eb256ac4dc97d04880e22deb92c8521b6 100644
--- a/CRC-HL2_EmitaCityRP.Emita_2025/spawnLootSystem.sqf
+++ b/CRC-HL2_EmitaCityRP.Emita_2025/spawnLootSystem.sqf
@@ -42,34 +42,43 @@
                                 _pool = _lootMilitary; _containers = _containersMilitary; _count = 10 + floor random 12; _spawnChance = 0.4;
                             };
                         };
                     };
                 };
 
                 if (!(_pool isEqualTo []) && {random 1 < _spawnChance}) then {
                     private _crateClass = selectRandom _containers;
                     private _box = _crateClass createVehicle _pos;
 
                     clearItemCargoGlobal _box;
                     clearWeaponCargoGlobal _box;
                     clearMagazineCargoGlobal _box;
                     clearBackpackCargoGlobal _box;
 
                     for "_i" from 1 to _count do {
                         _box addItemCargoGlobal [selectRandom _pool, 1];
                     };
 
                     if (_nameLower find "loot_military_" == 0 && {random 1 < 0.25}) then {
                         _box addItemCargoGlobal [selectRandom _lootEquipment, 1];
                     };
                     if (_nameLower find "loot_food_" == 0 && {random 1 < 0.20}) then {
                         _box addItemCargoGlobal [selectRandom _lootMedical, 1];
                     };
+                    _box addEventHandler ["Killed", {
+                        private _amount  = 2 + floor random 4;
+                        private _targets = allPlayers select { side _x == west && alive _x };
+                        {
+                            for "_i" from 1 to _amount do { _x addItem "VRP_HL_Token_Item"; };
+                        } forEach _targets;
+                        [format ["Loot cache destroyed. You receive %1 Tokens.", _amount]]
+                            remoteExec ["hintSilent", _targets apply { owner _x }];
+                    }];
                 };
 
                 deleteMarker _m;
             };
         } forEach _markers;
 
         sleep 10;
     };
 };
